### PR TITLE
[ISSUE #4733] Substitute e.printStackTrace() with log.error()

### LINKED ISSUE #4733: [Enhancement] Substitute `e.printStackTrace()` with `log.error()`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

image

`e.printStackTrace()` should be substituted with `log.error()` in these classes:

- org.apache.eventmesh.runtime.client.impl.PubClientImpl
- org.apache.eventmesh.openconnect.SourceWorker
- org.apache.eventmesh.common.ThreadWrapperTest

### Describe the solution you'd like

Take `org.apache.eventmesh.runtime.client.impl.PubClientImpl` as an example:

image

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct) *

### PR DESCRIPTION
**Pull request by scwlkq**
Fixes #4733 .

### Modifications

Substitute e.printStackTrace() with log.error()

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/client/impl/PubClientImpl.java [hunk 1]
```diff
@@ -73,7 +73,7 @@ public void close() {
             task.cancel(false);
             super.close();
         } catch (Exception e) {
-            e.printStackTrace();
+            log.error("PubClientImpl|{}|stopped!", clientNo);
         }
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/client/impl/PubClientImpl.java:77]: How about passing `e` and printing stacktrace here?

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

**Comment by scwlkq** [comment]
image
Is stopped suitable here? Or there may be other words

File: eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/client/impl/PubClientImpl.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public void close() {
             task.cancel(false);
             super.close();
         } catch (Exception e) {
-            e.printStackTrace();
+            log.error("PubClientImpl|{}|stopped!", clientNo, e);
         }
```
Comment (Pil0tXia) [inline eventmesh-runtime/src/test/java/org/apache/eventmesh/runtime/client/impl/PubClientImpl.java:77]: The logger here is in a catch block so the log message would better be:

```suggestion
        } catch (Exception e) {
            log.error("PubClientImpl|{}|close failed!", clientNo, e);
        }
```

**Review by Pil0tXia — CHANGES_REQUESTED** [review]

**Review by mxsm — APPROVED** [review]

**Review by Pil0tXia — APPROVED** [review]
LGTM

**Pull request merged**